plugin/core/helpers.py

```diff
@@ -0,0 +1,17 @@
+from threading import Timer
```

Comment:
Did you try using `sublime.set_timeout_async` instead of a `threading.Timer`?

According to the [docs](https://docs.python.org/3/library/threading.html#timer-objects), a Timer is a subclass of a Thread. So as it stands now a thread is constantly created and destroyed whenever the view changes. This seems wasteful to me. While on the other hand the `sublime.set_timeout_async` function runs a function after a timeout on a dedicated "async" thread (from what I can understand from the ST api). This "async" thread keeps on living forever and various tiny functions may be run on it.

Reply:
I will try it and let you know how it turns out :)

Reply:
Well I expected that the `set_time_out_async` return something like a **`timer`**, but it returns `None`. And the sublime api doesn't have a `clear_time_out_async` function that would allow me to cancel that given **`timer`**. 

So I don't seem to know how to implement this with a `set_time_out_async`. 
https://github.com/tomv564/LSP/blob/ba832ee73dcf5d26cef8e64193a71c2a87dcfc5c/plugin/core/helpers.py#L11

Reply:
Right, there's no way to cancel a timer with the ST api unfortunately.